File: Java/src/test/java/joseahernandez/fizzbuzz/FizzBuzzTest.java
package joseahernandez.fizzbuzz;

import org.junit.Test;
import static org.junit.Assert.*;

public class FizzBuzzTest {

    @Test
    public void fizzBuzzReturnTheNumberPassedAsArgument() {
        int number = 1;
        String result = FizzBuzz.fizzBuzz(number);

        assertEquals(String.valueOf(number), result);
    }

    @Test
    public void multiplesOfThreeReturnFizz() {
        int number = 3;
        String result = FizzBuzz.fizzBuzz(number);

        assertEquals("Fizz", result);
    }

    @Test
    public void multiplesOfFiveReturnBuzz() {
        int number = 5;
        String result = FizzBuzz.fizzBuzz(number);

        assertEquals("Buzz", result);
    }

    @Test
    public void multiplesOfThreeAndFiveReturnFizzBuzz() {
        int number = 15;
        String result = FizzBuzz.fizzBuzz(number);

        assertEquals("FizzBuzz", result);
    }
}

File: Kotlin/src/test/kotlin/joseahernandez/fizzbuzz/FizzBuzzTest.kt
package joseahernandez.fizzbuzz

import org.junit.Test
import kotlin.test.assertEquals

class FizzBuzzTest {
    private val fizzBuzz: FizzBuzz = FizzBuzz()

    @Test
    fun fizzBuzzReturnTheNumberPassedAsArgument() {
        val number = 1
        val result = fizzBuzz.fizzBuzz(number)

        assertEquals(number.toString(), result)
    }

    @Test
    fun multiplesOfThreeReturnFizz() {
        val number = 3
        val result = fizzBuzz.fizzBuzz(number)

        assertEquals("Fizz", result)
    }

    @Test
    fun multiplesOfFiveReturnBuzz() {
        val number = 5
        val result = fizzBuzz.fizzBuzz(number)

        assertEquals("Buzz", result)
    }

    @Test
    fun multiplesOfThreeAndFiveReturnFizzBuzz() {
        val number = 15
        val result = fizzBuzz.fizzBuzz(number)

        assertEquals("FizzBuzz", result)
    }
}
